### ISSUE TITLE
Add a rule to validate security in the servers

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2123]
**User story.**
As a user I want to know if the server has the correct security defined (that given security key has reference to the schemes defined in the `components.securitySchemes`).

**Is your feature request related to a problem?**
The following AsyncAPI document is invalid and Spectral should throw errors due to not defined security schemes inside one of server:

```yaml
asyncapi: '2.0.0',
servers:
  production:
    security:
    - not_defined: []
components:
  securitySchemes:
    petstore_auth:
      ...
```

**Additional context**
Part of #2100
